src/unity/lib/extensions/model_base.hpp

```diff
@@ -83,7 +83,7 @@ class EXPORT model_base: public cppipc::ipc_object_base {
    * Serializes the toolkit class. Must save the class to the file format
    * version matching that of get_version().
    */
-  virtual void save_impl(oarchive& oarc) const;
+  virtual void save_impl(oarchive& oarc) const {};
```

Comment:
Why do we want to provide an empty implementation here? This doesn't seem like the right behavior - if all subclasses are expected to implement this method, it should be pure virtual (`= 0`), or if there is some reason it can't be, perhaps it should throw an exception at runtime in case it somehow gets called.

Reply:
Eh, model_base isn't expected to be serializable, but ml_model base is.  We had to make a few of these changes for weird reasons.  I think the best update is to undo the removal of the pure virtual methods in ml_model.hpp.

Reply:
@hoytak I agree, putting back the pure virtual methods in ml_model.hpp sounds right. Should we do that as part of this PR?

Reply:
Thank you guys. A comment note added to the empty implementations as per offline discussion.